common/src/lib/java/dev/engine_room/flywheel/lib/model/baked/MeshEmitterManager.java

```diff
@@ -0,0 +1,78 @@
+package dev.engine_room.flywheel.lib.model.baked;
+
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import com.mojang.blaze3d.vertex.BufferBuilder;
+
+import it.unimi.dsi.fastutil.objects.Reference2ReferenceLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.objects.Reference2ReferenceMap;
+import it.unimi.dsi.fastutil.objects.ReferenceArrayList;
+import net.minecraft.client.renderer.RenderType;
+
+class MeshEmitterManager<T extends MeshEmitter> {
+	private static final RenderType[] CHUNK_LAYERS = RenderType.chunkBufferLayers().toArray(RenderType[]::new);
+
+	final Reference2ReferenceMap<RenderType, T> emitterMap = new Reference2ReferenceLinkedOpenHashMap<>();
+	private final ReferenceArrayList<BufferBuilder> bufferBuilders = new ReferenceArrayList<>();
+
+	@UnknownNullability
+	private ResultConsumer<?> resultConsumer;
+	private int nextBufferBuilderIndex;
+
+	MeshEmitterManager(BiFunction<Supplier<BufferBuilder>, RenderType, T> meshEmitterFactory) {
+		for (RenderType renderType : CHUNK_LAYERS) {
+			T emitter = meshEmitterFactory.apply(this::getOrCreateBufferBuilder, renderType);
```

Comment:
Each mesh emitter gets a separate function object here, but also I don't think the Supplier<BufferBuilder> needs to be a member variable of MeshEmitter.

You could store the buffer builder re-use logic in its own class, keep a final instance of it in this class, and pass it directly into MeshEmitter#getBuffer here: https://github.com/Engine-Room/Flywheel/pull/299/files#diff-1164194e4f2fd6cf77fb60e2b6cdbf10a46fbf79bbb990e3ecba404f29f3137dR61